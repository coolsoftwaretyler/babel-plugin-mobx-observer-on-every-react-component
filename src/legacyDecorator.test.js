import { expect, test, describe } from "bun:test";
import { transform } from "@babel/core";
import pluginSyntaxJsx from "@babel/plugin-syntax-jsx";
import pluginTransReactJsx from "@babel/plugin-transform-react-jsx";
import pluginSyntaxDecoratorsLegacy from "@babel/plugin-proposal-decorators";
import { autoObserverPlugin } from "./index.mjs";

const runTransformLegacy = (input) => {
    return transform(input, {
        plugins: [pluginSyntaxJsx, pluginTransReactJsx, [pluginSyntaxDecoratorsLegacy, { legacy: true }], autoObserverPlugin],
    });
};

describe("Mobx Observer Babel Plugin", () => {
    describe('legacy decorators', () => {
        test('do not get double wrapped', () => {
            const { code } = runTransformLegacy(`var _class;
let MyComponent = observer(_class = class MyComponent {
  render() {
    return /*#__PURE__*/React.createElement("div", null, "Hello World");
  }
}) || _class;`)
            expect(code).toMatchSnapshot()
        })

    })
});



