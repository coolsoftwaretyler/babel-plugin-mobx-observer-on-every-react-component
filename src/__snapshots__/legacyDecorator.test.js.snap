// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx Observer Babel Plugin legacy decorators do not get double wrapped 1`] = `
"import { observer } from "mobx-react";
var _class;
let MyComponent = observer(_class = class MyComponent {
  render() {
    return /*#__PURE__*/React.createElement("div", null, "Hello World");
  }
}) || _class;"
`;
